packages/wallet/src/communication/index.ts

```diff
@@ -0,0 +1,56 @@
+import { Commitment } from '../domain';
+import { messageRelayRequested } from 'magmo-wallet-client';
+
+export enum Strategy {
+  IndirectFunding = 'IndirectFunding',
+}
```

Comment:
It seems like `type Strategy = 'IndirectFunding' | ... ` might be superior here. Having a type makes it easier to combine these options or add more complex options (e.g. `IndirectFunding(with_buffer: 20%)`) later. VSCode gives you all the auto-complete goodness.

Reply:
There are, on the other hand, some downsides to a discriminated union:
You can do this:
```
export const INDIRECT = 'IndirectFundingStrategy'
export type INDIRECT = typeof INDIRECT;
export type FundingStrategy = INDIRECT;
```
which gets verbose, for no clear benefit if you don't have complex types.

Or you can do this: 
```
export type FundingStrategy = 'IndirectFundingStrategy';
```

in which case you need to do things like
```
const strategy = 'IndirectFundingStrategy' as 'IndirectFundingStrategy';
// or:
// const strategy: FundingStrategy = 'IndirectFundingStrategy';
const waitForStrategyResponse = states.waitForStrategyResponse(props);
```
because Typescript does not know that `strategy` was supposed to be of type `FundingStrategy`, and assigns it type `string`.

The enum approach has some other benefits:
- you can iterate over the values of an enum (perhaps useful in views)
- The values of an enum are tightly tied to the enum type itself: The only way to access `Strategy.IndirectFunding` is through the `Strategy` enum, while the string `'IndirectFundingStrategy'` can be used in any context that expects a string.

I've tried switching to the simple union type of anonymous strings.